### PR TITLE
Set role examples to v4 and add detail warnings

### PR DESCRIPTION
**Pull request by xacrimon**

### CODE REVIEW

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 1]
```diff
@@ -4,6 +4,12 @@ description: Moderated Sessions
 h1: Moderated Sessions
 ---
 
+<Details title="Details title" min="8.3" opened>
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:7]: I suggest condensing the text into something like,

```markdown
Moderated Sessions is only available in Teleport 8.3.0 and above. If you are using a previous version of Teleport, you must specify `version: v4` or earlier for role resources.
```

File: docs/pages/access-controls/reference.mdx [hunk 2]
```diff
@@ -41,9 +41,15 @@ $ tctl get roles
 
 A role definition looks like this:
 
+<Details title="Details title" min="8.3" opened>
+  The role example below uses `version: v4` for compatability reasons. Note that
+  the `join_sessions` and `require_session_join` fields are only supported on `version: v5` roles
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:46]:
```suggestion
  the `join_sessions` and `require_session_join` fields are only supported on `version: v5` roles,
```

File: docs/pages/access-controls/reference.mdx [hunk 3]
```diff
@@ -230,12 +236,12 @@ that are more appropriately scoped.
 
 ### Role versions
 
-There are currently two supported role versions: `v3` and `v5`. `v5` roles are
+There are currently two supported role versions: `v3` and `v4`. `v4` roles are
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:239]: Should we mention v5 in a third column here?

File: docs/pages/access-controls/reference.mdx [hunk 4]
```diff
@@ -230,12 +236,13 @@ that are more appropriately scoped.
 
 ### Role versions
 
-There are currently two supported role versions: `v3` and `v5`. `v5` roles are
+There are currently three supported role versions: `v3`, `v4` and `v5`. `v4` roles are
 completely backwards-compatible with `v3`, the only difference lies in the
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:240]:
```suggestion
completely backwards-compatible with `v3`. The only difference lies in the
```

File: docs/pages/access-controls/reference.mdx [hunk 5]
```diff
@@ -230,12 +236,13 @@ that are more appropriately scoped.
...
 default allow labels which will be applied to the role if they are not
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:241]:
```suggestion
default allow labels, which will be applied to the role if they are not
```

File: docs/pages/access-controls/reference.mdx [hunk 6]
```diff
@@ -230,12 +236,13 @@ that are more appropriately scoped.
...
-explicitly set.
+explicitly set. `v5` roles are also backwards compatible with `v4`. They add
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:242]:
```suggestion
explicitly set.

`v5` roles are also backwards compatible with `v4`. They add
```

**Review by ptgott — APPROVED** [review]
Made some minor suggestions but approved so you're not blocked.

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 1 above)
Comment (zmb3) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:7]: Don't forget to update the title!

Comment (xacrimon) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:7]: Good catch

**Review by zmb3 — COMMENTED** [review]
Feels like we should mention that v4 roles will be automatically migrated to v5 upon upgrading to teleport 8.3.0

**Comment by xacrimon** [comment]
> Feels like we should mention that v4 roles will be automatically migrated to v5 upon upgrading to teleport 8.3.0

@zmb3 Added a sentence about it but note that they aren't automatically migrated due to the RBAC poisoning mechanics for SSH.

**Comment by benarent** [comment]
> Feels like we should mention that v4 roles will be automatically migrated to v5 upon upgrading to teleport 8.3.0

I'm running a 8.0.0 and my roles are still on `v3` when did we introduce v4?

**Comment by xacrimon** [comment]
@benarent It's complicated, v4 roles came around during v6 or so I believe. That said, a lot of internal systems still rely on V3 behaviour which is why it's a default for many internal roles and other things.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 7]
```diff
@@ -4,6 +4,11 @@ description: Moderated Sessions
 h1: Moderated Sessions
 ---
 
+<Details title="Version Requirements" min="8.3" opened>
+  Moderated Sessions is only available in Teleport 8.3.0 and above.
+  If you are using a previous version of Teleport, you must specify `version: v4` or earlier for role resources.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:9]: This sentence makes it sound like if you use an older version of Teleport and specify v4 that you can still use moderated sessions.

How about something like:

```
Moderated Sessions require `v5` roles, which are only available in Teleport 8.3.0 and above.
```

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
Merged.

@xacrimon don't forget to make any corresponding updates to `master` before we cut `branch/v9` later this week.

**Pull request merged**